tests/templates/__init__.py

```diff
@@ -35,62 +36,103 @@ class Response(NamedTuple):
 
 class ImplRequest:
     """
-    Implementation of a org.freedesktop.impl.portal.Request object. Typically
-    this object needs to be merely exported:
+    Implementation of a ``org.freedesktop.impl.portal.Request`` object exposed
+    on the object path ``handle``.
 
-        >>> r = ImplRequest(mock, "org.freedesktop.impl.portal.Test", handle)
-        >>> r.export()
-
-    Where the test or the backend implementation relies on the Closed() method
-    of the ImplRequest, provide a callback to be invoked.
-
-        >>> r.export(close_callback=my_callback)
-
-    Note that the latter only works if the test invokes methods
-    asynchronously.
-
-    .. attribute:: closed
-
-        Set to True if the Close() method on the Request was invoked
+    The dbus method implementations need to be invoked asynchronously and the
+    async callbacks must be passed in ``cb_success`` and ``cb_error``.
 
+    The request either waits until it is closed by x-d-p (``wait_for_close``) or
+    responds to the request (``respond``).
     """
 
-    def __init__(self, mock: "dbusmock.DBusMockObject", busname: str, handle: str):
+    def __init__(
+        self,
+        mock: "dbusmock.DBusMockObject",
+        busname: str,
+        handle: str,
+        logger,
+        cb_success,
+        cb_error,
+    ):
         self.mock = mock
-        self.handle = handle
-        self.closed = False
-        self._close_callback: Optional[Callable] = None
-
         bus = mock.connection
         proxy = bus.get_object(busname, handle)
-        mock_interface = dbus.Interface(proxy, dbusmock.MOCK_IFACE)
+        self.mock_interface = dbus.Interface(proxy, dbusmock.MOCK_IFACE)
+        self.handle = handle
+        self.logger = logger
+        self.cb_success = cb_success
+        self.cb_error = cb_error
+
+    def respond(self, response, delay=0, done_cb=None):
```

Comment:
```suggestion
    def respond(self, response: Callable | Response, delay=0, done_cb=None):
```
would be a nice annotation here (I think that's the type)

Reply:
Yeah, I forgot to add typing annotations because mypy and the templates don't work well together so I excluded them.